src/SubscriptionService/SubscriptionService.cs

```diff
@@ -0,0 +1,492 @@
+    {
+        #region Fields
+
+        /// <summary>
+        /// The connection string
+        /// </summary>
+        private readonly String ConnectionString;
+
+        /// <summary>
+        /// The destroy lock
+        /// </summary>
+        private readonly Object DestroyLock = new Object();
+
+        /// <summary>
+        /// The event store connection
+        /// </summary>
+        private readonly IEventStoreConnection EventStoreConnection;
+
+        /// <summary>
+        /// The HTTP client
+        /// </summary>
+        private readonly HttpClient HttpClient;
+
+        /// <summary>
+        /// The lock object
+        /// </summary>
+        private readonly Object LockObject = new Object();
+
+        /// <summary>
+        /// The manual reset event
+        /// </summary>
+        private readonly ManualResetEvent ManualResetEvent = new ManualResetEvent(true);
+
+        /// <summary>
+        /// The subscriptions
+        /// </summary>
+        private readonly List<Subscription> Subscriptions;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SubscriptionService" /> class.
+        /// </summary>
+        /// <param name="subscriptions">The subscriptions.</param>
+        /// <param name="eventStoreConnection">The event store connection.</param>
+        /// <param name="username">The username.</param>
+        /// <param name="password">The password.</param>
+        public SubscriptionService(List<Subscription> subscriptions,
+                                   IEventStoreConnection eventStoreConnection,
+                                   String username = "admin",
+                                   String password = "changeit")
+        {
+            this.Subscriptions = subscriptions;
+            this.EventStoreConnection = eventStoreConnection;
+
+            // Wire up the event store connection events
+            this.EventStoreConnection.Connected += this.EventStoreConnection_Connected;
+            this.EventStoreConnection.Closed += this.EventStoreConnection_Closed;
+            this.EventStoreConnection.Reconnecting += this.EventStoreConnection_Reconnecting;
+            this.EventStoreConnection.ErrorOccurred += this.EventStoreConnection_ErrorOccurred;
+            this.EventStoreConnection.AuthenticationFailed += this.EventStoreConnection_AuthenticationFailed;
+            this.EventStoreConnection.Disconnected += this.EventStoreConnection_Disconnected;
+
+            // Cache the user credentials
+            this.DefaultUserCredentials = new UserCredentials(username, password);
+
+            // Default value at the moment less than the default Event Store Retry of 10 seconds
+            Int32 httpRequestTimeout = 8;
+
+            // Create our Http Client to publish the message to the consumer endpoint
+            this.HttpClient = new HttpClient();
```

Comment:
We don't know the endpoint(s) yet so we have lost the benefit of any potential connection pooling.

Reply:
A valid point will raise an issue for this.